### PR TITLE
make lorawan-encoding support 1.0.4 

### PR DESCRIPTION
**Pull request by lucasgranberg**
This cleans up maccommandcreator and adds ADR fields.

### CODE REVIEW

File: lorawan-encoding/src/maccommandcreator.rs [hunk 1]
```diff
@@ -612,11 +660,13 @@ impl RXTimingSetupReqCreator {
 /// let creator = lorawan::maccommandcreator::RXTimingSetupAnsCreator::new();
 /// let res = creator.build();
 /// ```
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct RXTimingSetupAnsCreator {}
 
 impl_mac_cmd_creator_boilerplate!(RXTimingSetupAnsCreator, 0x08);
 
-#[derive(Debug, PartialEq, Eq)]
```
Comment (lthiery) [inline lorawan-encoding/src/maccommandcreator.rs:619]: Is there reasoning in removing PartialEq, Eq?

File: lorawan-encoding/src/parser.rs [hunk 2]
```diff
@@ -1083,6 +1083,16 @@ impl FCtrl {
         FCtrl(bytes, uplink)
     }
 
+    /// Set ADR ACK requested.
+    pub fn set_adr_ack_req(&mut self) {
+        self.0 |= 0b1 << 6;
```
Comment (plaes) [inline lorawan-encoding/src/parser.rs:1088]: Just use `1 << 6`,

File: lorawan-encoding/src/parser.rs [hunk 3]
```diff
@@ -1083,6 +1083,16 @@ impl FCtrl {
...
+    }
+
+    /// Set ADR enabled.
+    pub fn set_adr(&mut self) {
+        self.0 |= 0b1 << 7;
```
Comment (plaes) [inline lorawan-encoding/src/parser.rs:1093]: Ditto.

File: lorawan-encoding/src/maccommandcreator.rs (hunk 1 above)
Comment (lucasgranberg) [reply lorawan-encoding/src/maccommandcreator.rs:619]: my reasoning was that they were implemented only for some commands and not all. I also don't see a reason for comparing two maccommandcreator objects. I can add them back if needed, Ill try to check in git blame why they are there in the first place.

Comment (lucasgranberg) [reply lorawan-encoding/src/maccommandcreator.rs:619]: Turns out I am to blame. I added them half-assedly when I first added the 1.0.4 commands.

**Pull request merged**